src/main/java/jewellery/inventory/service/SaleService.java

```diff
@@ -164,10 +180,117 @@ private void deleteSaleIfProductsIsEmpty(Sale sale) {
 
   private ProductReturnResponseDto validateSaleAfterReturnProduct(
       Sale sale, Product productToReturn) {
-    if (sale.getProducts().isEmpty()) {
+    if (sale.getProducts().isEmpty() && sale.getResources().isEmpty()) {
       return productService.getProductReturnResponseDto(null, productToReturn);
     }
     return productService.getProductReturnResponseDto(
         saleMapper.mapEntityToResponseDto(sale), productToReturn);
   }
+
+  private ResourceReturnResponseDto validateSaleAfterReturnResource(
+      Sale sale, PurchasedResourceInUser resourceToReturn) {
+    if (sale.getResources().isEmpty() && sale.getProducts().isEmpty()) {
+      return saleMapper.mapToResourceReturnResponseDto(resourceToReturn, null);
+    }
```

Comment:
We can do the following. We'd need to add a null check inside saleMapper::mapEntityToResponseDto(Sale sale) to not throw NullPointer
```suggestion
    if (sale.getResources().isEmpty() && sale.getProducts().isEmpty()) {
      sale = null
    }
```

Reply:
Done!